packages/app/src/components/approve/ApproveToken.tsx

```diff
@@ -0,0 +1,64 @@
+import { FC, MouseEventHandler } from "react";
+import { Button, Container, Tooltip, OverlayTrigger, Spinner } from "react-bootstrap";
+import { BigNumberish, constants, Contract } from "ethers";
+import { useContractFunction } from "@usedapp/core";
+
+type Props = {
+  symbol: string;
+  hasAllowance: boolean;
+  spender: string;
+  tokenAddress: Contract;
+  amount?: BigNumberish;
+};
+
+const ApproveToken: FC<Props> = ({ symbol, spender, hasAllowance, tokenAddress, amount }) => {
```

Comment:
style nit 

It might be interesting to have some order in the props from more general to more specific 

e.g. symbol, hasAllowance, tokenAddress, spender, amount 

If there is some logical ordering it can reduce developer error in the future and leads to a codebase that is just a little more pleasant to work with.